Snowflake.Data.Tests/ChunkStreamingParserTest.cs

```diff
@@ -0,0 +1,160 @@
+        }
+
+        [TearDown]
+        public void AfterTest()
+        {
+            SFConfiguration.Instance().ChunkParserVersion = ChunkParserVersionDefault; // Return to default version
+        }
+
+        [Test]
+        [Ignore("ChunkStreamingParserTest")]
+        public void ChunkStreamingParserTestDone()
+        {
+            // Do nothing;
+        }
+
+        public IChunkParser getParser(string data)
+        {
+            byte[] bytes = Encoding.UTF8.GetBytes(data);
+            Stream stream = new MemoryStream(bytes);
+            return ChunkParserFactory.Instance.GetParser(stream);
+        }
+
+        [Test]
+        public async Task TestParsingEmptyChunk()
+        {
+            // Create sample data for parser
+            string data = "[ ]";
+            IChunkParser parser = getParser(data);
+
+            SFResultChunk chunk = new SFResultChunk(new string[0, 0]);
+
+            await parser.ParseChunk(chunk);
+
+            Assert.AreEqual(0, chunk.rowSet.GetLength(0)); // Check row length
+            Assert.AreEqual(0, chunk.rowSet.GetLength(0)); // Check col length
+            Assert.Throws<IndexOutOfRangeException>(() => chunk.ExtractCell(0, 0).SafeToString());
+        }
+
+        [Test]
+        public async Task TestParsingEmptyArraysInChunk()
+        {
+            // Create sample data for parser
+            string data = "[ [],  [] ]";
+            IChunkParser parser = getParser(data);
+
+            SFResultChunk chunk = new SFResultChunk(new string[2, 0]);
+
+            await parser.ParseChunk(chunk);
+
+            Assert.AreEqual(2, chunk.rowSet.GetLength(0)); // Check row length
+            Assert.AreEqual(0, chunk.rowSet.GetLength(1)); // Check col length
+            Assert.Throws<IndexOutOfRangeException>(() => chunk.ExtractCell(0, 0).SafeToString());
+        }
+
+        [Test]
+        public async Task TestParsingNonJsonArrayChunk()
+        {
+            // Create a sample data using data not contained in an array
+            string data = "[ \"1\", \"1.234\", \"abcde\" ]";
+            IChunkParser parser = getParser(data);
+
+            SFResultChunk chunk = new SFResultChunk(new string[1, 3]);
+
+            await parser.ParseChunk(chunk);
+
+            // ChunkStreamingParser is able to parse one row that is not inside of an array
+            Assert.AreEqual("1", chunk.ExtractCell(0, 0).SafeToString());
+            Assert.AreEqual("1.234", chunk.ExtractCell(0, 1).SafeToString());
+            Assert.AreEqual("abcde", chunk.ExtractCell(0, 2).SafeToString());
+        }
+
+        [Test]
+        public void TestParsingNonJsonArrayAndOneArray()
+        {
+            // Create a sample data using non-array data and an array
+            string data = "[ \"1\", \"1.234\", \"abcde\", [\"2\", \"5.678\", \"fghi\"] ]";
+            IChunkParser parser = getParser(data);
+
+            SFResultChunk chunk = new SFResultChunk(new string[1, 3]);
```

Comment:
if 2nd dimension would be higher that the test fails which means that parser didn't throw an exception.
Is this expected behaviour?

Reply:
I'm honestly not sure if that's the expected behavior. The parser class was created before we started and I couldn't find any tests/docs about it

I'm assuming it was never meant to parse non-array data based on this: https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Core/SFBlockingChunkDownloaderV3.cs#L216C5-L216C5